Source/WTF/wtf/text/StringCommon.h

```diff
@@ -967,16 +967,24 @@ inline bool equalIgnoringASCIICase(ASCIILiteral a, ASCIILiteral b)
 }
 
 template<typename ElementType>
-inline void copyElements(ElementType* __restrict destination, const ElementType* __restrict source, size_t length)
+inline void copyElements(std::span<ElementType> destinationSpan, std::span<const ElementType> sourceSpan)
 {
-    if (length == 1)
+    ASSERT(destinationSpan.size() >= sourceSpan.size());
+    auto* __restrict destination = destinationSpan.data();
+    auto* __restrict source = sourceSpan.data();
+    if (sourceSpan.size() == 1)
         *destination = *source;
-    else if (length)
-        std::memcpy(destination, source, length * sizeof(ElementType));
+    else if (!sourceSpan.empty())
+        std::memcpy(destination, source, sourceSpan.size_bytes());
 }
 
-inline void copyElements(uint16_t* __restrict destination, const uint8_t* __restrict source, size_t length)
+inline void copyElements(std::span<uint16_t> destinationSpan, std::span<const uint8_t> sourceSpan)
 {
+    ASSERT(destinationSpan.size() >= sourceSpan.size());
+    auto* __restrict destination = destinationSpan.data();
```

Comment:
Ditto. It seems that we should have a debug assert that destination & sourceSpan don't overlap.